dev/changelog/42.0.0.md

```diff
@@ -0,0 +1,418 @@
+- Bump rstest from 0.17.0 to 0.22.0 in /datafusion-cli [#12413](https://github.com/apache/datafusion/pull/12413) (dependabot[bot])
+- fix tpc-h parquet setting to respect global options [#12405](https://github.com/apache/datafusion/pull/12405) (XiangpengHao)
+- Bump dirs from 4.0.0 to 5.0.1 in /datafusion-cli [#12411](https://github.com/apache/datafusion/pull/12411) (dependabot[bot])
+- Allow using dictionary arrays as filters [#12382](https://github.com/apache/datafusion/pull/12382) (adriangb)
+- Add support for Utf8View, Boolean, Date32/64, int32/64 for writing hive style partitions [#12283](https://github.com/apache/datafusion/pull/12283) (Omega359)
+- Bump env_logger from 0.9.3 to 0.11.5 in /datafusion-cli [#12410](https://github.com/apache/datafusion/pull/12410) (dependabot[bot])
+- Check window functions by str for with_column [#12431](https://github.com/apache/datafusion/pull/12431) (timsaucer)
+- Fix incorrect OFFSET during LIMIT pushdown. [#12399](https://github.com/apache/datafusion/pull/12399) (wiedld)
+- Fix: upper case qualifier wildcard bug [#12426](https://github.com/apache/datafusion/pull/12426) (JasonLi-cn)
+- Fix: Internal error in regexp_replace() for some StringView input [#12203](https://github.com/apache/datafusion/pull/12203) (devanbenz)
+- Automate sqllogictest for StringView (for one function, `substr`) [#12433](https://github.com/apache/datafusion/pull/12433) (2010YOUY01)
+- Update concat_ws scalar function to support Utf8View [#12309](https://github.com/apache/datafusion/pull/12309) (devanbenz)
+- Bump serve-static and express in /datafusion/wasmtest/datafusion-wasm-app [#12434](https://github.com/apache/datafusion/pull/12434) (dependabot[bot])
+- Minor: add err on `create` `temporary` table [#12439](https://github.com/apache/datafusion/pull/12439) (hailelagi)
+- Minor: Add a test for version() function [#12441](https://github.com/apache/datafusion/pull/12441) (alamb)
+- Handle case-sensitive identifier when decorrelating predicate subquery [#12443](https://github.com/apache/datafusion/pull/12443) (goldmedal)
+- Bump send and express in /datafusion/wasmtest/datafusion-wasm-app [#12447](https://github.com/apache/datafusion/pull/12447) (dependabot[bot])
+- Add PartialOrd for the DF subfields/structs for the WindowFunction expr [#12421](https://github.com/apache/datafusion/pull/12421) (ngli-me)
+- Making avro_to_arrow::schema::to_arrow_schema public [#12452](https://github.com/apache/datafusion/pull/12452) (ameyc)
+- Bump rustyline from 11.0.0 to 14.0.0 in /datafusion-cli [#12407](https://github.com/apache/datafusion/pull/12407) (dependabot[bot])
+- Minor: add `ListingOptions::with_file_extension_opt` [#12461](https://github.com/apache/datafusion/pull/12461) (alamb)
+- Improve PhysicalExpr and Column documentation [#12457](https://github.com/apache/datafusion/pull/12457) (alamb)
+- fix length error with `array_has` [#12459](https://github.com/apache/datafusion/pull/12459) (samuelcolvin)
```

Comment:
this is a bug fix.

Reply:
The changelog script categorizes PRs based on GitHub labels and/or [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) prefixes, so PRs with the `bug` label or a commit message starting with `fix:` would get categorized as bug fixes.

Reply:
Script for reference: https://github.com/apache/datafusion/blob/main/dev/release/generate-changelog.py

Reply:
@samuelcolvin  Let me know if you want to add labels to any PRs and I can regenerate the changelog in this PR

Reply:
Ok, no worries, I don't have permissions to edit labels.

No need to change it for me.

Thanks as ever for datafusion.

Reply:
I used to review all the PRs when generating changelogs, but with ~300 per release, it got a bit much. It looks like we haven't documented how the changelog generation works in the contributor guide. I will add that to my TODO list.